feathr_project/feathr/client.py

```diff
@@ -361,6 +379,14 @@ def get_offline_features(self,
             output_path: output path of job, i.e. the observation data with features attached.
             execution_configuratons: a dict that will be passed to spark job when the job starts up, i.e. the "spark configurations". Note that not all of the configuration will be honored since some of the configurations are managed by the Spark platform, such as Databricks or Azure Synapse. Refer to the [spark documentation](https://spark.apache.org/docs/latest/configuration.html) for a complete list of spark configurations.
         """
+        feature_queries = feature_query if isinstance(feature_query, List) else [feature_query]
+        feature_names = []
+        for feature_query in feature_queries:
+            for feature_name in feature_query.feature_list:
+                feature_names.append(feature_name)
+        preprocessing_pyudf_manager = _PreprocessingPyudfManager()
```

Comment:
Looks like _PreprocessingPyudfManager is stateless, may rename to _PreprocessingPyudfUtils?